R/add_dd.R

```diff
@@ -0,0 +1,314 @@
+#' Add Delay Discharge to working file
+#'
+#' @param data The input data frame
+#' @param year The year being processed
+#'
+#' @return A data frame linking delay discharge cohorts
+#' @export
+#'
+#' @family episode file
+add_dd <- function(data, year) {
+  year_param <- year
+
+  data <- data %>%
+    dplyr::mutate(
+      # remember to revoke the cij_end_date with dummy_cij_end
+      cij_start_date_lower = .data$cij_start_date - lubridate::days(1),
+      cij_end_date_upper = cij_end_date + lubridate::days(1),
+      cij_end_month = last_date_month(cij_end_date),
+      is_dummy_cij_start = is.na(cij_start_date) & !is.na(cij_end_date),
+      dummy_cij_start = dplyr::if_else(
+        is_dummy_cij_start,
+        lubridate::as_date("1900-01-01"),
+        cij_start_date_lower
+      ),
+      is_dummy_cij_end = !is.na(cij_start_date) & is.na(cij_end_date),
+      dummy_cij_end = dplyr::if_else(
+        is_dummy_cij_end,
+        lubridate::today(),
+        cij_end_month
+      )
+    )
```

Comment:
```suggestion
  data <- data %>%
    dplyr::mutate(
      # remember to revoke the cij_end_date with dummy_cij_end
      cij_start_date_lower = .data$cij_start_date - lubridate::days(1),
      cij_end_date_upper = .data$cij_end_date + lubridate::days(1),
      cij_end_month = last_date_month(.data$cij_end_date),
      is_dummy_cij_start = is.na(.data$cij_start_date) & !is.na(.data$cij_end_date),
      dummy_cij_start = dplyr::if_else(
        is_dummy_cij_start,
        lubridate::as_date("1900-01-01"),
        .data$cij_start_date_lower
      ),
      is_dummy_cij_end = !is.na(.data$cij_start_date) & is.na(.data$cij_end_date),
      dummy_cij_end = dplyr::if_else(
        is_dummy_cij_end,
        lubridate::today(),
        cij_end_month
      )
    )
```
I think we need to add `.data$` throughout to address the notes in the build check